packages/plugin/webpack/src/WebpackConfig.ts

```diff
@@ -163,114 +180,122 @@ export default class WebpackConfigGenerator {
     );
   }
 
-  async getPreloadConfigForEntryPoint(entryPoint: WebpackPluginEntryPointLocalWindow | WebpackPluginEntryPointPreloadOnly): Promise<Configuration> {
-    if (!entryPoint.preload) {
-      return {};
+  async getRendererConfig(entryPoints: WebpackPluginEntryPoint[]): Promise<Configuration[]> {
+    const entryPointsForTarget = {
+      web: [] as (WebpackPluginEntryPointLocalWindow | WebpackPluginEntryPoint)[],
+      electronRenderer: [] as (WebpackPluginEntryPointLocalWindow | WebpackPluginEntryPoint)[],
+      electronPreload: [] as (WebpackPluginEntryPointPreloadOnly | WebpackPluginEntryPointLocalWindow)[],
+      sandboxedPreload: [] as (WebpackPluginEntryPointPreloadOnly | WebpackPluginEntryPointLocalWindow)[],
+    };
+    for (const entry of entryPoints) {
+      if (entry.nodeIntegration ?? this.pluginConfig.renderer.nodeIntegration) {
+        if (isPreloadOnly(entry)) {
+          entryPointsForTarget.electronPreload.push(entry);
+        } else {
+          entryPointsForTarget.electronRenderer.push(entry);
+          if (isLocalWindow(entry) && entry.preload) {
+            entryPointsForTarget.electronPreload.push(entry);
+          }
+        }
+      } else {
+        if (isPreloadOnly(entry)) {
+          entryPointsForTarget.sandboxedPreload.push(entry);
+        } else {
+          entryPointsForTarget.web.push(entry);
+          if (isLocalWindow(entry) && entry.preload) {
+            entryPointsForTarget.sandboxedPreload.push(entry);
+          }
+        }
+      }
     }
+    const rendererConfigPromises = [
+      this.buildRendererConfig(entryPointsForTarget.web, RendererTarget.Web),
+      this.buildRendererConfig(entryPointsForTarget.electronRenderer, RendererTarget.ElectronRenderer),
+      this.buildRendererConfig(entryPointsForTarget.electronPreload, RendererTarget.ElectronPreload),
+      this.buildRendererConfig(entryPointsForTarget.sandboxedPreload, RendererTarget.SandboxedPreload),
+    ];
```

Comment:
```suggestion
    const rendererConfigPromises = await Promise.all([
      this.buildRendererConfig(entryPointsForTarget.web, RendererTarget.Web),
      this.buildRendererConfig(entryPointsForTarget.electronRenderer, RendererTarget.ElectronRenderer),
      this.buildRendererConfig(entryPointsForTarget.electronPreload, RendererTarget.ElectronPreload),
      this.buildRendererConfig(entryPointsForTarget.sandboxedPreload, RendererTarget.SandboxedPreload),
    ]);
```

Let's do the promise awaiting inline, we shouldn't be assigning promises to a value as it potentially let's us experience unhandled rejections with future refactors